### PR TITLE
Fix bug with flickering recipes, history, and pantry pages

### PR DESCRIPTION
**Pull request by phuang00**

### CODE REVIEW

File: client/components/Recipe/RecipesPage.vue [hunk 1]
```diff
@@ -1,68 +1,92 @@
...
-        />
+  <main>
+    <AddRecipeForm />
+    <hr>
+    <h2 class="display-4">Recipes</h2>
+    <FilterRecipeForm />
+    <br>
+    <div class="font-italic">
+      <h2 v-if="$store.state.keyword && $store.state.ingredients.length">
+        Displaying recipes with keyword {{ $store.state.keyword }} and ingredient(s) {{ ingredients.join(',') }}
```
Comment (jlin00) [inline client/components/Recipe/RecipesPage.vue]: Should `ingredients.join(',') ` be `$store.state.ingredients.join(', ')`?

**Review by jlin00 — CHANGES_REQUESTED** [review]
Small change requested

File: client/components/Recipe/RecipesPage.vue (hunk 1 above)
Comment (phuang00) [reply client/components/Recipe/RecipesPage.vue]: fixed

**Review by jlin00 — APPROVED** [review]

**Comment by phuang00** [comment]
Put recipes back into store and updating way of fetching

File: client/components/common/BlockForm.vue [hunk 2]
```diff
@@ -521,7 +522,7 @@ export default {
         }
         
         if (this.refreshRecipes) {
-          this.$store.commit('refreshRecipes');
+          await this.$store.commit('fetchRecipes', this.showLoading ? this.showLoading : false);
```
Comment (jlin00) [inline client/components/common/BlockForm.vue]: Where is this.showLoading coming from in block form?

**Review by jlin00 — CHANGES_REQUESTED** [review]

File: client/components/common/BlockForm.vue (hunk 2 above)
Comment (phuang00) [reply client/components/common/BlockForm.vue]: updated, it's legacy from prev idea i had in mind

**Review by jlin00 — APPROVED** [review]

**Pull request merged**